circuits/cpp/src/aztec3/circuits/hash.hpp

```diff
@@ -112,6 +114,41 @@ typename NCT::fr root_from_sibling_path(typename NCT::fr const& leaf,
     return node; // root
 }
 
+/**
+ * @brief Calculate the Merkle tree root from the sibling path and leaf.
+ *
+ * @details The leaf is hashed with its sibling, and then the result is hashed
+ * with the next sibling etc in the path. The last hash is the root.
+ *
+ * @tparam NCT Operate on NativeTypes or CircuitTypes
+ * @tparam N The number of elements in the sibling path
+ * @param leaf The leaf element of the Merkle tree
+ * @param leafIndex The index of the leaf element in the Merkle tree
+ * @param siblingPath The nodes representing the merkle siblings of the leaf, its parent,
+ * the next parent, etc up to the sibling below the root
+ * @return The computed Merkle tree root.
+ *
+ * TODO need to use conditional assigns instead of `ifs` for circuit version.
+ *      see membership.hpp:check_subtree_membership (left/right/conditional_assign, etc)
+ */
+template <typename NCT, size_t N>
+typename NCT::fr root_from_sibling_path(typename NCT::fr const& leaf,
+                                        typename NCT::fr const& leafIndex,
+                                        std::array<typename NCT::fr, N> const& siblingPath)
+{
+    auto node = leaf;
+    uint256_t index = leafIndex;
```

Comment:
Converting from NCT::fr to uint256_t, is this ok?

Reply:
I think this line is able to implicitly use the user-defined conversion operator (defined in the `field.hpp` class in bberg):
`operator uint256_t() const noexcept {...}`
I'm not too familiar with such things, but yes, I think this is ok to do.